components/leave_private_channel_modal/leave_private_channel_modal.tsx

```diff
@@ -96,10 +96,26 @@ export default class LeavePrivateChannelModal extends React.PureComponent<Props,
             );
         }
 
+        if (this.state.channel?.type === Constants.GM_CHANNEL) {
+            title = (
+                <FormattedMessage
+                    id='leave_group_message_modal.title'
+                    defaultMessage='Leave this Group Message?'
+                />
+            );
+
+            message = (
+                <FormattedMessage
+                    id='leave_group_message_modal.message'
+                    defaultMessage="You won't be able to rejoin this group unless you are invited again"
+                />
+            );
+        }
+
         const buttonClass = 'btn btn-danger';
         const button = (
             <FormattedMessage
-                id='leave_private_channel_modal.leave'
+                id={`${this.state.channel?.type === Constants.GM_CHANNEL ? 'leave_group_message_modal' : 'leave_private_channel_modal'}.leave`}
```

Comment:
IMO the differentiation between private and group message could be done right at the beginning of the component. All these conditions seem a bit over the top. Maybe even set the `id`s and `defaultMessage`s right in the beginning. This would certainly improve readability